Otus-DevOps-2020-02/DmitryKorlas_microservices
# Docker 3

# Completed the homework №16


## What were done:
 - add microservices structure of app
 - Add Dockerfiles and run microservices app. Describe steps in README.md
 - (the task *): run containers using custom network aliases. Setup communication between them using env variables.
 - (the task **): optimize an image of src/ui:
   - Created multi-stage build for reducing an image size. See result in **Dockerfile.3.0** file.
   - Used lightweight **ruby:2.6.5-alpine** image vs initial heavy **ruby:2.2**.
   - The final sizes of **ui** image: v1.0=**770MB**, v2.0=**447MB**, v3.0=**67.2MB**


## How to run the project:
 - See README.md for details. In short terms - build 4 images: **post:1.0**, **comment:1.0**, **ui:3.0** and mongo, then start it.

## How to check, that it works correctly:
 - visit http://34.78.251.183:9292/ (available temporary). See **IP** from output of command `docker-machine ls` for receiving the actual IP address. 
 - add some post and comment using WEB UI
 - re-start containers `docker kill $(docker ps -aq)`, to start see README.md for details
    ```sh
    docker run -d --network=reddit \
      --network-alias=post_db \
      --network-alias=comment_db \
      -v reddit_db:/data/db mongo:latest 
   ```
  - visit WEB UI again - it should display previously added post and comment

## PR checklist:
 - [x] Added a label with the number pf homework
